### ISSUE TITLE
[HOLD for payment 2023-09-29] [$500]  Assign task - Colon (:) added to completed/reopened task messages when copy pasting

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#26753]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. login to staging new dot
2. Go to any chat with a task
3. Copy the completed or reopened task message using the copy to clipboard icon
4. Paste the copied content in the composer and observe that a colon is added after completed or reopend when pasting

## Expected Result:
Pasted content should be identical to copied content

## Actual Result:
Pasted content has colon

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.3.63-0

**Reproducible in staging?:** Y

**Reproducible in production?:** Y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/0513ebc1-23aa-474a-a8db-55016d754dcc

https://github.com/Expensify/App/assets/115492554/0270bbef-614d-41ed-9f91-5f0f521313f2

**Expensify/Expensify Issue URL:**

**Issue reported by:** ~@sofoniasN~ @oleksandr-pantsyr (reported first [here](https://github.com/Expensify/App/issues/26571))

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692985594255829

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012810233ffe197834</li>
        <li>Upwork Job ID: 1699809589610123264</li>
        <li>Last Price Increase: 2023-09-07</li>
<li>Automatic offers: </li>
<ul>
<li>Ollyws | Reviewer | 26589564</li>
<li>sofoniasN | Reporter | 26589566</li>
<li>oleksandr-pantsyr | Contributor | 26974844</li>
</ul></li>
    </ul>
</details>

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- Colon (:) added to completed/reopened task messages when copy pasting

### What is the root cause of that problem?
- Currently, the copied text is from the originalMessage that has the colon (:): https://github.com/Expensify/App/blob/f67bf9f88c08fe2cebe16202d1244299a9640ea3/src/pages/home/report/ContextMenu/ContextMenuActions.js#L189
- But the displayed task action message is from: ````${taskStatusText} ${taskReportName}```` that does not has the colon (:): 
https://github.com/Expensify/App/blob/f67bf9f88c08fe2cebe16202d1244299a9640ea3/src/components/ReportActionItem/TaskAction.js#L38-L63

### What changes do you think we should make in order to solve the problem?
- We can change the way to get the copied text to the same way that used to get the task action message

### What alternative solutions did you explore? (Optional)
- NA

**Comment by BhuvaneshPatil** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

### What is the root cause of that problem?
We use originalMessage to get the text to be copied here -
https://github.com/Expensify/App/blob/b95c1dd22474fb48929e78a102c1121c1e6e4360/src/pages/home/report/ContextMenu/ContextMenuActions.js#L190

and it looks like this 
<img width="294" alt="Screenshot 2023-09-05 at 5 00 36 PM" src="https://github.com/Expensify/App/assets/27822551/c959f327-128d-4c84-ac37-c4058223c767">

Observe it contains semicolon.
but while rendering it we display that separately(prop translation)
https://github.com/Expensify/App/blob/b95c1dd22474fb48929e78a102c1121c1e6e4360/src/components/ReportActionItem/TaskAction.js#L57-L63

### What changes do you think we should make in order to solve the problem?
There are two approaches that to solve this -
- We either change the `TaskAction` and add colon there
```js
<Text style={[styles.chatItemMessage, styles.colorMuted]}>{`${taskStatusText}: ${taskReportName}`}</Text>
```
- We change the logic for copy action here -
https://github.com/Expensify/App/blob/b95c1dd22474fb48929e78a102c1121c1e6e4360/src/pages/home/report/ContextMenu/ContextMenuActions.js#L190

For this we can create a new method in ReportActionUtils - `getTaskActionText(actionName)` it will return translated key for all the `CONST.REPORT.ACTIONS.TYPE` and then use it on above mentioned line while copying the text
### What alternative solutions did you explore? (Optional)

**Comment by zanyrenney** [comment]
Yep, reproducible here - adding external.

**Comment by luacmartins** [comment]
Agreed with C+ assessment.

**Comment by DylanDylann** [comment]
@Ollyws @luacmartins The PR https://github.com/Expensify/App/pull/27119 ready for review

**Comment by pecanoro** [comment]
This got a bit chaotic, mostly because https://github.com/Expensify/App/issues/26571 was opened so reported before this one so the other one should take priority. That means that we won't be paying the reporting bonus for this one.

However, we DO need to fix all cases in one single PR. That means that either @DylanDylann, you fix all problems in your PR or the original contributor does it.

**Comment by DylanDylann** [comment]
@pecanoro I will fix all cases in my PR. My current PR also fix https://github.com/Expensify/App/issues/26571

**Comment by pecanoro** [comment]
Awesome, glad to hear! I will close the other one!

@zanyrenney The original reporter is @oleksandr-pantsyr since this was a duplicate with the same root cause.

**Comment by DylanDylann** [comment]
The PR is out of scope to fix some related issues. It caused the timeline extending beyond 3 days

**Comment by pecanoro** [comment]
I don't think we have sent any offers yet, we will handle after the regression period.

**Comment by zanyrenney** [comment]
@Ollyws requires payment [offer](https://www.upwork.com/nx/wm/offer/26589564) (Reviewer) $500
@DylanDylann requires payment $500
@oleksandr-pantsyr requires payment for reporting bonus $50

**Comment by zanyrenney** [comment]
waiting on @DylanDylann and @oleksandr-pantsyr to accept job offers.

**Comment by DylanDylann** [comment]
@pecanoro @Ollyws Could you help to check this comment https://github.com/Expensify/App/issues/26753#issuecomment-1728986092 ? Many thanks

**Comment by zanyrenney** [comment]
Sorry @DylanDylann this doesn't qualify for the bonus, the rules are set on the days taken to merge. As with all issues, we try to be as fair as possible but are operating within the guidelines set.

**Comment by zanyrenney** [comment]
All paid out!
